digdag-standards/src/main/java/io/digdag/standards/operator/td/TdWaitOperatorFactory.java

```diff
@@ -0,0 +1,170 @@
+                TDJob jobInfo = job.getJobInfo();
+                TDJob.Status status = jobInfo.getStatus();
+                logger.debug("poll job status: {}: {}", existingJobId.get(), jobInfo);
+
+                // Wait some more if the job is not yet finished
+                if (!status.isFinished()) {
+                    throw TaskExecutionException.ofNextPolling(JOB_STATUS_API_POLL_INTERVAL, ConfigElement.copyOf(state));
+                }
+
+                // Fail the task if the job failed
+                if (status != SUCCESS) {
+                    String message = jobInfo.getCmdOut() + "\n" + jobInfo.getStdErr();
+                    throw new TaskExecutionException(message, ConfigElement.empty());
+                }
+
+                // Fetch the job output to see if the query condition has been fulfilled
+                logger.debug("fetching poll job result: {}", existingJobId.get());
+                boolean done = fetchJobResult(job);
+
+                // If the query condition was not fulfilled, go back to sleep.
+                if (!done) {
+                    state.remove(JOB_ID);
+                    throw TaskExecutionException.ofNextPolling(pollInterval, ConfigElement.copyOf(state));
+                }
+
+                // The query condition was fulfilled, we're done.
+                return TaskResult.empty(request);
+            }
+        }
+
+        private String startJob(TDOperator op)
+        {
+            TDJobRequest req = new TDJobRequestBuilder()
+                    .setType(engine)
+                    .setDatabase(op.getDatabase())
+                    .setQuery(query)
+                    .setRetryLimit(jobRetry)
+                    .setPriority(priority)
+                    .setScheduledTime(request.getSessionTime().getEpochSecond())
+                    .createTDJobRequest();
+
+            TDJobOperator job = op.submitNewJob(req);
+            logger.info("Started {} job id={}:\n{}", engine, job.getJobId(), query);
+
+            return job.getJobId();
+        }
+
+        private boolean fetchJobResult(TDJobOperator job)
+        {
+            Optional<ArrayValue> firstRow = job.getResult(ite -> ite.hasNext() ? Optional.of(ite.next()) : Optional.absent());
+
+            // There must be at least one row in the result for the wait condition to be fulfilled.
+            if (!firstRow.isPresent()) {
+                return false;
+            }
+
+            ArrayValue row = firstRow.get();
+            if (row.size() < 1) {
+                throw new TaskExecutionException("Got empty row in result of query", ConfigElement.empty());
+            }
+
+            // Wait condition is fulfilled if the first column is not NULL and not FALSE.
+            Value firstCol = row.get(0);
+            return !firstCol.isNilValue() &&
+                    !(firstCol.isBooleanValue() && !firstCol.asBooleanValue().getBoolean());
```

Comment:
I think some people who are familiar with MySQL will use "select 1" instead of "select true" by mistake because MySQL doesn't have boolean type.
How about adding this condition as well?:
`!(firstCol.isIntegerValue() && !firstCol.asIntegerValue().equals(BigInteger.valueOf(0)))`